cardano-db/test/Test/IO/Cardano/Db/Insert.hs

```diff
@@ -48,6 +49,19 @@ insertFirstTest =
     bid1 <- insertBlockChecked $ (\b -> b { blockPreviousId = Just bid0 }) (blockOne slid)
     assertBool (show bid0 ++ " == " ++ show bid1) (bid0 /= bid1)
 
+insertTwice :: IO ()
+insertTwice =
+  runDbNoLogging $ do
+    slid <- insertSlotLeader testSlotLeader
+    bid <- insertBlockChecked (blockZero slid)
+    let adaPots = adaPotsZero bid
+    _ <- insertAdaPots adaPots
+    Just pots0 <- queryAdaPots bid
+    -- Insert with same Unique key, different first field
+    _ <- insertAdaPots (adaPots {adaPotsSlotNo = 1 + adaPotsSlotNo adaPots})
+    Just pots0' <- queryAdaPots bid
+    assertBool (show (adaPotsSlotNo pots0) ++ " /= " ++ show (adaPotsSlotNo pots0'))
+      (adaPotsSlotNo pots0 == adaPotsSlotNo pots0')
```

Comment:
AdaPots triggers the bug, because its first field (Slot) is not part of the Unique Key.